### PR TITLE
Add support for docker run --tmpfs flag.

### PR DESCRIPTION
**Pull request by sllawap**
This adds support for the docker run --tmpfs flag as described in https://github.com/docker/compose/issues/2778

### CODE REVIEW

**Comment by sllawap** [comment]
@GordonTheTurtle -- should be all set now. Sorry for the oversight; first time contributor!

**Comment by dato** [comment]
I tried to test this PR, because I wanted to check whether syntax like `/tmp:size=10M` or `/home/foo/tmp:uid=NNN` would work.

However, I could not make the PR run. I think support for tmpfs is needed in docker-py, maybe?

Consider the following _docker-compose.yml_:

```yaml
version: "2"

services:
  test:
    image: busybox
    tmpfs: /run
    network_mode: none
```

I run: 

```
$ env PYTHONPATH=. bin/docker-compose up test
```

It fails with:

```pytb
Traceback (most recent call last):
  File "bin/docker-compose", line 3, in <module>
    main()
  File ".../src/compose/compose/cli/main.py", line 56, in main
    command.sys_dispatch()
  File ".../src/compose/compose/service.py", line 648, in _get_container_host_config
    tmpfs=options.get('tmpfs'),
  File ".../Library/Python/Envs/compose/lib/python2.7/site-packages/docker/api/container.py", line 141, in create_host_config
    return utils.create_host_config(*args, **kwargs)
TypeError: create_host_config() got an unexpected keyword argument 'tmpfs'
```

I looked at the [AppVeyor logs](https://ci.appveyor.com/project/docker/compose/build/master-1328), but I didn’t see the [`test_tmp()`](https://github.com/docker/compose/pull/2978/files#diff-ebf882803c8eac5a345368609d88cd80R874) test run, only the simpler [`test_tmpfs_option()`](https://github.com/docker/compose/pull/2978/files#diff-02a52ec1e2850804d6071dcb286cfebcR1184). I don’t really know if it the test was supposed to catch it or not.

File: compose/config/config.py [hunk 1]
```diff
@@ -725,6 +729,7 @@ def merge_service_dicts(base, override, version):
         'external_links',
         'ports',
         'volumes_from',
+        'tmpfs',
```
Comment (dnephin) [inline compose/config/config.py]: Since this can be a list or a string it needs to be in the list with `dns` and `dns_search`.

**Comment by dnephin** [comment]
Thanks for the PR!

I've left one comment about a fix.
You'll also need to rebase on master, squash to one commit, and make sure that commit is signed.

If this requires a specific pre-release version of `docker-py` you can update the required version in `requirements.txt`. We're currently using a pinned version, so you might not need to make any changes there.

**Comment by dnephin** [comment]
Carried in #3137

**Comment by sllawap** [comment]
dnephin@ sorry for neglecting this, I was on vacation. Thank you for pushing this through!

**Pull request closed without merging**